src/test/operators/export_binary_test.cpp

```diff
@@ -52,12 +52,30 @@ class OperatorsExportBinaryTest : public BaseTest {
   const std::string filename = test_data_path + "export_test.bin";
 };
 
+class OperatorsExportBinaryMultiEncodingTest : public OperatorsExportBinaryTest,
```

Comment:
Just some cases that we should make sure to cover:
- single chunk
- multiple chunks
- All-NULL
- Repeated values (for RLE)
- Sorted / unsorted values

Reply:
Since RLE segments are usually created from value segments and value segments guarantee that stored values for NULL-attributed values are default-constructed `T{}`, it's rather complicated to construct such a table as in the example. I did so by directly assigning the values to the value vector of the segment, since all other access methods prevent the value vector from containing actual values if the null value property is set. Consequently someone must be very malicious to construct such a table. But in case it happens, it's tested now.

Reply:
Doesn't that happen with the current encoder? To me, it looks like for NULL values, it continues the current run without looking at the value in the input segment:

https://github.com/hyrise/hyrise/blob/9d274347689191dc5faa619800c4475b7ce7c83e/src/lib/storage/run_length_segment/run_length_encoder.hpp#L47-L53

Reply:
Ah, no, I misunderstood that code. In any case, that is what should happen. I'll open an issue.

Reply:
Ok, I was entirely mistaken, likely confused by this code in run_length_segment.hpp:

```
As in value segments, null values are represented as an additional boolean vector.
```

I would remove that sentence and change a sentence a couple of lines above (including a typo fix):

```
Each run is represented by the value and the number of **occurrences. Runs can also be NULL, in which case `values` is undefined (but likely `T{}`) and `null_values` is set to true for that run.**

Reply:
Moving this to #1928 so that we can find an end here.